app/src/main/kotlin/compiler/intermediate/generators/DefaultFunctionDetailsGenerator.kt

```diff
@@ -113,20 +114,22 @@ data class DefaultFunctionDetailsGenerator(
                 CFGLinkType.UNCONDITIONAL,
                 genWrite(
                     param.value,
-                    IFTNode.Add(
-                        IFTNode.RegisterRead(Register.RBP), // add 2 to account old RBP and return address
-                        IFTNode.Const((param.index + 2) * memoryUnitSize.toLong())
+                    IFTNode.MemoryRead(
+                        IFTNode.Add(
+                            IFTNode.RegisterRead(Register.RBP), // add 2 to account old RBP and return address
+                            IFTNode.Const((param.index + 2) * memoryUnitSize.toLong())
+                        )
                     ),
                     true
                 ),
             )
         }
 
         // backup callee-saved registers
-        for (register in calleeSavedRegistersWithoutRSPAndRBP.reversed())
+        for ((hardware_reg, backup_reg) in calleeSavedRegistersWithoutRSPAndRBP zip calleeSavedBackupRegisters)
```

Comment:
:snake: → :camel:

Reply:
It took me some time (plus help from my brother) to understand this comment, but of course you're right.